evalml/pipelines/utils.py

```diff
@@ -134,15 +134,12 @@ def make_pipeline(X, y, estimator, problem_type, custom_hyperparameters=None):
     if custom_hyperparameters and not isinstance(custom_hyperparameters, dict):
         raise ValueError(f"if custom_hyperparameters provided, must be dictionary. Received {type(custom_hyperparameters)}")
 
-    hyperparameters = custom_hyperparameters
     base_class = _get_pipeline_base_class(problem_type)
+    name = f"{estimator.name} w/ {' + '.join([component.name for component in preprocessing_components])}"
+    if parameters is None:
+        parameters = {}
 
-    class GeneratedPipeline(base_class):
-        custom_name = f"{estimator.name} w/ {' + '.join([component.name for component in preprocessing_components])}"
-        component_graph = complete_component_graph
-        custom_hyperparameters = hyperparameters
-
-    return GeneratedPipeline
+    return base_class(complete_component_graph, custom_name=name, parameters=parameters, custom_hyperparameters=custom_hyperparameters)
 
 
 def make_pipeline_from_components(component_instances, problem_type, custom_name=None, random_seed=0):
```

Comment:
Also honestly no longer really useful, but I think it's okay to leave for now and delete in separate PR.